### DISCUSSION TITLE
Identity Provider Username Mapper

### DISCUSSION THREAD

**Discussion by ericxbenoit** [keycloak/keycloak#21928] [Q&A]
I use Keycloak with GitHub as Identity Provider. Previously I was configuring a mapper of type 'Username Template Importer' with the value ${CLAIM.email}. By default, it created a user in Keycloak with email as username. Now it leaves the field empty and displays a form to enter it manually. Is there a way to use email as username automatically without using 'registrationEmailAsUsername' option? Thanks

**Comment by ericxbenoit** [comment]
The previous version was 12.0.4 and the new version I am using is 21.1.1.
I retried with the old version and it is still working fine but with the new version of Keycloak it is like I can't access CLAIM.. any ideas?

**Reply by giriparus** [reply]
I am facing a similar issue with microsoft. It works just fine with google but not microsoft. Did you ever get a solution for it?
I am using keycloak version 21.1.2
